front/src/modules/ui/view-bar/components/FilterDropdownEntitySearchSelect.tsx

```diff
@@ -72,6 +74,46 @@ export const FilterDropdownEntitySearchSelect = ({
     }
   };
 
+  const [filterDropdownSearchInput] = useRecoilScopedState(
+    filterDropdownSearchInputScopedState,
+    ViewBarRecoilScopeContext,
+  );
+
+  const isAllEntitySelected =
+    !entitiesForSelect.selectedEntities[0] &&
+    filterDropdownSelectedEntityId === ' ';
+
+  const isAllEnititySelectShown =
+    !!filterDefinitionUsedInDropdown?.selectAllLabel &&
+    !!filterDefinitionUsedInDropdown?.SelectAllIcon &&
+    (isAllEntitySelected ||
+      filterDefinitionUsedInDropdown?.selectAllLabel
+        .toLocaleLowerCase()
+        .includes(filterDropdownSearchInput.toLocaleLowerCase()));
+
+  const handleAllEntitySelectClick = () => {
+    if (
+      !filterDefinitionUsedInDropdown ||
+      !filterDefinitionUsedInDropdown.selectAllLabel
+    ) {
+      return;
+    }
+    if (isAllEntitySelected) {
+      removeFilter(filterDefinitionUsedInDropdown.key);
+      setFilterDropdownSelectedEntityId(null);
+    } else {
+      upsertFilter({
+        displayValue: filterDefinitionUsedInDropdown.selectAllLabel,
+        key: filterDefinitionUsedInDropdown.key,
+        operand: FilterOperand.IsNot,
+        type: filterDefinitionUsedInDropdown.type,
+        value: ' ',
```

Comment:
This is a hacky way to select all assignees. 
Here I am selecting all assignees where assigneeId != " "
Shall I create a new FilterOperand "All" for selecting all items?

Reply:
I actually think this "All assignee" is just another way to select none.
Can we use the emptyLabel from `SingleEntitySelectBase` and pass "All assignee". And when the empty option is selected, completely remove the filter in `FilterDropdownEntitySearchSelect`?

Reply:
Currently, there is no direct way to do a select-all query, there are only two filter operands - Is, IsNot

Here I am calling assigneeId != " ",  it will match with everything that is not equal to  a single space, hence returning all tasks for every assignee.

Reply:
Can we just get rid of the filter completely in this case?

Reply:
actually this is already implemented if you re-click on an the already selected option:
```
if (clickedOnAlreadySelectedEntity) {
      removeFilter(filterDefinitionUsedInDropdown.key);
      setFilterDropdownSelectedEntityId(null);
    } 
```

We should also trigger the same logic if you click on the "All assignee" option (and I recommend using the empty option to do that)

Reply:
@charlesBochet 
currently, the empty filter behavior is to show empty tasks as in the video below :

https://github.com/twentyhq/twenty/assets/54364088/d56b739a-aa79-490a-b257-b67b4c1cc8a1

I have tried to retain the empty tasks view and created all assignee task selection

do you mean when no filter is selected I should display all tasks instead of empty tasks view ?

Reply:
It was what I was meaning above but I am unsure now.
For tasks assignee, we do not allow to select no assignee while editing a tasks. However, it would be convenient for other filters to have an option to select "all" vs selecting "none".

So I think what we want is to pass an:
- allIcon
- allLabel
to our SingleEntitySelectBase (similar to empty behavior)

When all is selected, add a filter that would be a "isNotNull".

Reply:
Regarding the props you pass to SingleEntitySelectBase:
SelectAllIcon, ==> ok
  selectAllLabel, ==> ok
  isAllEntitySelected, ==> ok
  isAllEnititySelectShown, ==>let's remove and let's compute it based on if selectAllLabel and selectAllIcons are present
  onAllEntitySelected, ==> ok

Reply:
mmmh this looks non straight forward at all looking at the codebase :p

Reply:
> Regarding the props you pass to SingleEntitySelectBase: SelectAllIcon, ==> ok selectAllLabel, ==> ok isAllEntitySelected, ==> ok isAllEnititySelectShown, ==>let's remove and let's compute it based on if selectAllLabel and selectAllIcons are present onAllEntitySelected, ==> ok

as per discussion with @Bonapara https://github.com/twentyhq/twenty/issues/1077#issuecomment-1739560850 
I need to factor in search input text as well, hence cannot compute `isAllEnititySelectShown` base on selectAllLabel & selectAllIcons alone

Reply:
I think we need to introduce an internal state isAllSelected in FilterDropdownEntitySearchSelect
Then introduce a new isNotNull operand in ViewFilterOperand
Looks not trivial!

Reply:
> mmmh this looks non straight forward at all looking at the codebase :p we might need heavy refactoring here :p

Sorry I am not able to understand this, I can see your comments one after the other. I don't understand what you are referring to

Reply:
I'm saying that our current implementation of the filters looks very intricated and this leads to a complex implementation of the feature you are trying to do, which is not a problem but will require more work than I expected originally

Reply:
> > Regarding the props you pass to SingleEntitySelectBase: SelectAllIcon, ==> ok selectAllLabel, ==> ok isAllEntitySelected, ==> ok isAllEnititySelectShown, ==>let's remove and let's compute it based on if selectAllLabel and selectAllIcons are present onAllEntitySelected, ==> ok
> 
> as per discussion with @Bonapara [#1077 (comment)](https://github.com/twentyhq/twenty/issues/1077#issuecomment-1739560850) I need to factor in search input text as well, hence cannot compute `isAllEnititySelectShown` base on selectAllLabel & selectAllIcons alone

I see, good point

Reply:
So to summarize, your PR is correct, except for the hacky part. (=! ' ')
You'll need to introduce a isNotNull capability on our filters

Reply:
On it!

Reply:
Thank you! Happy to help if needed, I will keep an eye on github :)

Reply:
@charlesBochet
1. Added local state for isAllEntitySelected.
1. Created a new FilterOperand InNotNull and used it here for selecting all assignees

Submitting again for review.